system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.cpp

```diff
@@ -33,40 +33,40 @@ PeriodicMeasurementNode::PeriodicMeasurementNode(
   const std::chrono::milliseconds publish_period,
   const bool clear_measurements_on_publish)
 : Node(name),
-  measurement_period_(measurement_period),
   publishing_topic_(publishing_topic),
-  measurement_timer_(nullptr),
-  publish_timer_(nullptr),
+  measurement_period_(measurement_period),
   publish_period_(publish_period),
-  clear_measurements_on_publish_(clear_measurements_on_publish)
-{}
+  clear_measurements_on_publish_(clear_measurements_on_publish) {}
 
 bool PeriodicMeasurementNode::setupStart()
 {
+  if (publisher_ == nullptr) {
+    publisher_ = create_publisher<MetricsMessage>(publishing_topic_, 10);
+  }
+
   if (!measurement_timer_) {
-    RCLCPP_DEBUG(this->get_logger(), "setupStart: creating measurement_timer_");
+    RCLCPP_DEBUG(get_logger(), "setupStart: creating measurement_timer_");
 
     measurement_timer_ = this->create_wall_timer(
-      measurement_period_,
-      std::bind(&PeriodicMeasurementNode::performPeriodicMeasurement, this));
-
+      measurement_period_, [this]() {this->performPeriodicMeasurement();});
+    measurement_start_ = this->now();
   } else {
     RCLCPP_WARN(this->get_logger(), "setupStart: measurement_timer_ already exists!");
   }
 
-  if (!publish_timer_ &&
-    publish_period_ != PeriodicMeasurementNode::DEFAULT_PUBLISH_WINDOW)
-  {
-    RCLCPP_DEBUG(this->get_logger(), "setupStart: creating publish_timer_");
+  if (!publish_timer_ && publish_period_ != PeriodicMeasurementNode::INVALID_PUBLISH_WINDOW) {
```

Comment:
All those if-s are error-proned, let's refactor as folow:

```c++
setupStart() {
  // reject if already started
  if (hasStarted_) {
    throw ...;
  }
  // check our assertions
  assert(publish_period_ != PeriodicMeasurementNode::INVALID_PUBLISH_WINDOW);
  assert(!publisher_);
  assert(...);

  // do stuff - NO IF STATEMENTS HERE
  publisher_ = create_publisher<...>(...);
}
```

Reply:
Removed `if` statements and added `assert()` statements.